propagators/opentelemetry-propagator-aws-xray/test/AWSXRayPropagator.test.ts

```diff
@@ -0,0 +1,282 @@
+  SpanContext,
+  TraceFlags,
+} from '@opentelemetry/api';
+import { TraceState } from '@opentelemetry/core';
+
+import { AWSXRAY_TRACE_ID_HEADER, AWSXRayPropagator } from '../src';
+
+describe('AWSXRayPropagator', () => {
+  const xrayPropagator = new AWSXRayPropagator();
+  const TRACE_ID = '8a3c60f7d188f8fa79d48a391a778fa6';
+  const SPAN_ID = '53995c3f42cd8ad8';
+  const SAMPLED_TRACE_FLAG = TraceFlags.SAMPLED;
+  const NOT_SAMPLED_TRACE_FLAG = TraceFlags.NONE;
+
+  let carrier: { [key: string]: unknown };
+
+  beforeEach(() => {
+    carrier = {};
+  });
+
+  describe('.inject()', () => {
+    it('should inject sampled context', () => {
+      const spanContext: SpanContext = {
+        traceId: TRACE_ID,
+        spanId: SPAN_ID,
+        traceFlags: SAMPLED_TRACE_FLAG,
+      };
+      xrayPropagator.inject(
+        setSpanContext(ROOT_CONTEXT, spanContext),
+        carrier,
+        defaultTextMapSetter
+      );
+
+      assert.deepStrictEqual(
+        carrier[AWSXRAY_TRACE_ID_HEADER],
+        'Root=1-8a3c60f7-d188f8fa79d48a391a778fa6;Parent=53995c3f42cd8ad8;Sampled=1'
+      );
+    });
+
+    it('should inject not sampled context', () => {
+      const spanContext: SpanContext = {
+        traceId: TRACE_ID,
+        spanId: SPAN_ID,
+        traceFlags: NOT_SAMPLED_TRACE_FLAG,
+      };
+      xrayPropagator.inject(
+        setSpanContext(ROOT_CONTEXT, spanContext),
+        carrier,
+        defaultTextMapSetter
+      );
+
+      assert.deepStrictEqual(
+        carrier[AWSXRAY_TRACE_ID_HEADER],
+        'Root=1-8a3c60f7-d188f8fa79d48a391a778fa6;Parent=53995c3f42cd8ad8;Sampled=0'
+      );
+    });
+
+    it('should inject with TraceState', () => {
+      const traceState = new TraceState();
+      traceState.set('foo', 'bar');
+      const spanContext: SpanContext = {
+        traceId: TRACE_ID,
+        spanId: SPAN_ID,
+        traceFlags: SAMPLED_TRACE_FLAG,
+        traceState: traceState,
+      };
+      xrayPropagator.inject(
+        setSpanContext(ROOT_CONTEXT, spanContext),
+        carrier,
+        defaultTextMapSetter
+      );
+
+      // TODO: assert trace state when the propagator supports it
+      assert.deepStrictEqual(
+        carrier[AWSXRAY_TRACE_ID_HEADER],
+        'Root=1-8a3c60f7-d188f8fa79d48a391a778fa6;Parent=53995c3f42cd8ad8;Sampled=1'
+      );
+    });
+
+    it('inject without spanContext - should inject nothing', () => {
+      xrayPropagator.inject(ROOT_CONTEXT, carrier, defaultTextMapSetter);
+
+      assert.deepStrictEqual(carrier, {});
+    });
+
+    it('inject default invalid spanContext - should inject nothing', () => {
+      xrayPropagator.inject(
+        setSpanContext(ROOT_CONTEXT, INVALID_SPAN_CONTEXT),
+        carrier,
+        defaultTextMapSetter
+      );
+
+      assert.deepStrictEqual(carrier, {});
+    });
+  });
+
+  describe('.extract()', () => {
+    it('extract nothing from context', () => {
+      // context remains untouched
+      assert.deepStrictEqual(
```

Comment:
```suggestion
      assert.strictEqual(
```
we should verify that we return the original object not one which looks like the original.